### PR TITLE
[processor/k8sattributes] Move k8sattr.fieldExtractConfigRegex.disallow feature gate to Beta

### LINKED ISSUE #25128: [processor/k8sattributes] Deprecate and remove FieldExtractConfig.Regex config option
The configuration interface of the k8sattributes receiver is over-complicated. This is an attempt to simplify it a bit.
 
`FieldExtractConfig` have `KeyRegex` and `Regex` options, which is pretty confusing. The `Regex` config option doesn't seem to be very important. I can be easily replaced with a transform processor with `replace_pattern` and `replace_match` functions. Therefore, I suggest:
1. Deprecate `FieldExtractConfig.Regex` option with instructions on how to replace it with the transform processor
2. Delete the option after 5 releases.

### PR DESCRIPTION
**Pull request by odubajDT**
#### Link to tracking issue
Fixes #25128

### CODE REVIEW

**Comment by odubajDT** [comment]
@TylerHelmuth @ChrsMark please review

File: .chloggen/k8sattributeprocessor-fg.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+note: "Move k8sattr.fieldExtractConfigRegex.disallow feature gate to Beta."
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [25128]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
+
```
Comment (ChrsMark) [inline .chloggen/k8sattributeprocessor-fg.yaml:19]: nit: 
```suggestion
  - Disable the `k8sattr.fieldExtractConfigRegex.disallow` feature gate to get the old behavior.
```

**Review by ChrsMark — APPROVED** [review]
LGTM

ref to initial plan: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33411/files#diff-8403376807ff74e7592002db4982bed0ead6639c8bb105780aef75418fd3d24eR10

File: .chloggen/k8sattributeprocessor-fg.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+note: "Move k8sattr.fieldExtractConfigRegex.disallow feature gate to Beta."
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [25128]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
+  - Disable the `k8sattr.fieldExtractConfigRegex.disallow` feature gate to get the old behavior.
```
Comment (odubajDT) [inline .chloggen/k8sattributeprocessor-fg.yaml:19]:
```suggestion
subtext: "Disable the `k8sattr.fieldExtractConfigRegex.disallow` feature gate to get the old behavior."
```

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**